modules/lib/tsconfig.build.json

```diff
@@ -1,9 +1,11 @@
-{
+{ // This file is used to build typescript declaration files
```

Comment:
I hope you delete such comments before merging, as they are pretty obvious and only pollute the code. More to that, comments are often forgotten to update after a change in the project concept or structure.

Reply:
> pretty obvious and only pollute the code
I disagree, I find such comments very useful.
Not everyone is a wizard when it comes to tsconfig.json.
It wasn't clear to me how the multiple tsconfig files worked, before I had studied the "whole" build setup.

These files aren't even part of the finished npm packages.
So I don't see the problem.

Reply:
**TL;DR:** Project-specific comments should be placed in one place, while technical comments become unnecessary once you learn how things work by reading the docs or talking with your team.

I believe we should maintain a certain level of expertise. For example, there are many things in Gradle build that aren't obvious at first glance, but we don't comment much there. You just go and learn it. If you didn't know something before, well, now you do.

Comments related to the build process can be placed in the `BUILDING.md` file, where the entire process can be explained, not just a small part in one of the files.

As for technical comments, like the one about the "target" property, as I said, we expect a certain level of expertise (I mean any developer in any project). If a developer doesn't know something, they go and read the docs and learn it. I expect that you, I, and everyone else working with this file knows how it works. If they don't, then why should they be making changes based on a few comments?

Reply:
Good luck when you get older and your brain starts forgetting... FIFO

Reply:
TL;DR Applies more to BUILDING.md than the file I'm directly working on.

Reply:
Also who I going to write and maintain BUILDING.md, and how would I know it exists? Should it be pointed to from all the files what it documents? Developers are notoriously "bad" at writing and maintaining documentation. Commenting seems less formal.

Reply:
No offense meant. I just find that excessive comments can be more of a distraction than a help, often cluttering the code without adding real value.

As for the BUILDING.md file, it can be placed next to the README, so it shouldn’t be hard to find. However, even with comments scattered across different files, understanding how things work might still require switching between them and reading all those comments, which doesn’t guarantee a clear understanding. If the goal is to provide a brief explanation of how things work, it’s better to do it in one place. A simple step-by-step guide, like those usually found under the "Usage"/"Installation" sections of many frameworks and tools, would be more effective, in my opinion. It may not be the task for this issue, of course.

Comments can also become outdated. It's rarely a problem to have comments in our code, but things get complicated with the configurations. For example, some properties in the TSConfig are now deprecated (and you even left a note there), but who’s responsible for updating or removing those comments when things no longer work the way they used to?  It falls on the same developers, and often, that doesn’t work well.

Reply:
> you even left a note there

No, I didn't :)

Reply:
That was a Monty Python reference btw :)
https://www.youtube.com/watch?v=uLlv_aZjHXc